rtwcli/rtw_cmds/rtw_cmds/workspace/verbs.py

```diff
@@ -12,11 +12,374 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+"""
+workspace concept.
+
+§§§ general configuration: "~/.ros_team_workspace/config.yaml
+alternative_ros_rolling_location=""
+team_license="Apache License 2.0"
+team_repository_server="https://github.com"
+team_private_config_path=""
+
+§§§ ros related params: "~/.ros_team_workspace/ros_config.yaml
+  - ros_env:
+    - ros_domain_id: 34
+
+§§§ workspaces: "~/.ros_team_workspace/workspaces.yaml"
+<workspace_name>:
+  - base_ws:
+  - ...
+  - docker_tag:
+  - ros_env:
+    - ros_domain_id: 1
+
+workspace name prefix is defined as first letters of the path and last folder if not "workspace":
+- /home/daniela/workspace/kuka_rolling_ws -> hdw__kuka_rolling_ws
+- /home/daniela/workspace/mojin/pid_ws -> hdw_mojin__pid_ws
+"""
+
+import copy
+import dataclasses
+import os
+import re
+import shutil
+import questionary
+from typing import Any, Dict, List
+from rtwcli.command.info import ROS_TEAM_WS_VARIABLES
+from rtwcli.helpers import (
+    create_file_and_write,
+    create_file_if_not_exists,
+    load_yaml_file,
+    write_to_yaml_file,
+)
 from rtwcli.verb import VerbExtension
+from send2trash import send2trash
+
+WORKSPACES_PATH = os.path.expanduser("~/.ros_team_workspace/workspaces.yaml")
+USE_WORKSPACE_SCRIPT_PATH = os.path.expanduser(
+    "~/ros_team_workspace/scripts/environment/setup.bash"
+)
+WORKSPACES_KEY = "workspaces"
+ROS_TEAM_WS_RC_PATH = os.path.expanduser("~/.ros_team_ws_rc")
+WORKSPACES_PATH_BACKUP = os.path.expanduser("~/.ros_team_workspace/workspaces_backup.yaml")
```

Comment:
We should work with backups similarly as Manual programmed, this by adding multiple versions of the files. Maybe adding those into a subfolder to not clutter the main folder would be good. We can then think about removing those files later if necessary.